packages/frontend-v2/components/AddCourseModal/AddCourseModal.tsx

```diff
@@ -65,15 +69,15 @@ export const AddCourseModal: React.FC<AddCourseModalProps> = ({
     const updatedSelectedCourses = [...selectedCourses];
 
     // grab any coreqs of the course that haven't already been selected/added to the term
-    const coreqs = (await getRequiredCourseCoreqs(course, catalogYear)).filter(
-      (coreq) => {
-        const isAlreadySelected = selectedCourses.find((selectedCourse) =>
-          isEqualCourses(selectedCourse, coreq)
-        );
-        const isAlreadyAdded = isCourseInCurrTerm(coreq);
-        return !(isAlreadyAdded || isAlreadySelected);
-      }
-    );
+    const coreqs = isAutoSelectCoreqs
+      ? (await getRequiredCourseCoreqs(course, catalogYear)).filter((coreq) => {
+          const isAlreadySelected = selectedCourses.find((selectedCourse) =>
+            isEqualCourses(selectedCourse, coreq)
+          );
+          const isAlreadyAdded = isCourseAlreadyAdded(coreq);
+          return !(isAlreadyAdded || isAlreadySelected);
+        })
+      : [];
```

Comment:
A little confused as to why the parameter is option here but that might just be me not being used to this syntax?

Reply:
if we don't want to be auto selecting co-req(which is the case when adding transfer courses), we can skip fetching it and just make coreqs an empty array